pytorch_blade/pytorch_blade/compiler/jit/torch/constant_propagation.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <memory>
+
+namespace torch {
+namespace jit {
+struct Graph;
+struct Block;
+struct Node;
+struct Value;
+} // namespace jit
+} // namespace torch
+
+namespace torch {
+namespace blade {
+
+using torch::jit::Block;
+using torch::jit::Graph;
+using torch::jit::Node;
+using torch::jit::Value;
+
+// Runs constant propagation on all objects unless ignore_custom_classes is
+// specified as true, in which case user defined classes are skipped.  This is
+// useful to prevent early fusion of packing operations, which end up lowering
+// away information about their constructors (e.g. packed::linear_clamp_prepack
+// and prepacked::conv2d_clamp_prepack)
+// Returns True if the pass made a change to the graph
+bool ConstantPropagation(std::shared_ptr<Graph>& graph);
```

Comment:
What's the main difference with torch's ?

Reply:
just removed const tensor folding from the original PyTorch code, because torch-mlir has some error on lowering with const tensor value, it seems to be related to dynamo tracing, this PR is a work around to avoid folding the const tensor value, we need to fix it in the future and using original const_propgation pass.

Reply:
I think the main consideration is to make const folding work with FakeTensor.